exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -171,15 +173,42 @@ static StackdriverV2ExporterHandler createWithStub(
   static StackdriverV2ExporterHandler createWithCredentials(
       String projectId,
       Credentials credentials,
-      Map<String, io.opencensus.trace.AttributeValue> fixedAttributes)
+      Map<String, io.opencensus.trace.AttributeValue> fixedAttributes,
+      @javax.annotation.Nullable Duration deadline)
       throws IOException {
-    TraceServiceSettings traceServiceSettings =
+    TraceServiceSettings.Builder builder =
         TraceServiceSettings.newBuilder()
             .setCredentialsProvider(
-                FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")))
-            .build();
+                FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")));
+    if (deadline != null) {
+      // We only use the batchWriteSpans API in this exporter.
+      builder
+          .batchWriteSpansSettings()
+          .setSimpleTimeoutNoRetries(org.threeten.bp.Duration.ofMillis(deadline.toMillis()));
+    } else {
+      /*
+       * Default retry settings for Stackdriver Trace client is:
+       * settings =
+       *   RetrySettings.newBuilder()
+       *       .setInitialRetryDelay(Duration.ofMillis(100L))
+       *       .setRetryDelayMultiplier(1.2)
+       *       .setMaxRetryDelay(Duration.ofMillis(1000L))
+       *       .setInitialRpcTimeout(Duration.ofMillis(30000L))
+       *       .setRpcTimeoutMultiplier(1.5)
+       *       .setMaxRpcTimeout(Duration.ofMillis(60000L))
+       *       .setTotalTimeout(Duration.ofMillis(120000L))
+       *       .build();
+       *
+       * Override the default settings with settings that don't retry.
+       */
+      builder
+          .batchWriteSpansSettings()
+          .setRetryableCodes() // Sets to empty so no retryable code will be used.
+          // RetrySettings.newBuilder().build() returns settings with no retries.
+          .setRetrySettings(RetrySettings.newBuilder().build());
```

Comment:
What is the default Deadline if non provided by the user?

Reply:
Total timeout is 120000 ms (2 min). See https://github.com/census-instrumentation/opencensus-java/pull/1881/files#diff-2f7e8b7487a68c644aef32903d35ca13R199.

Reply:
So the deadline will be setInitialRpcTimeout(60sec) because no retries? Can we explicitly set that? Can we also make it 10sec?

Reply:
> So the deadline will be setInitialRpcTimeout(60sec) because no retries? 

Apologies I forgot to mention 2min is the default deadline **before** this PR. After this PR if users don't set deadline, the default settings will be no retries and no timeout (see https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/retrying/RetrySettings.java#L137-L148). So if the initial RPC failed, all spans are immediately dropped.

> Can we explicitly set that? Can we also make it 10sec?

Yes and yes. Made the change in https://github.com/census-instrumentation/opencensus-java/pull/1881/commits/62aa27a4e47a9fe14f4cdfffa9b91b7655b24de9.